Kei okada's - get default planner from param

sachinchitta opened the pull request.

k-okada committed:
add getHandle to move_group_interface

k-okada committed:
motion_planning_frame_planning: use /default_planner_config parma to specify default planning algorithm

julienking committed:
fixing conflicts, renaming variable

sachinchitta commented:
@rhaschke - this param is only used to set the default planner for when the Rviz plugin gets brought up - currently it comes up as "unspecified". Assigning default planners per group would be useful but it should happen separately from this PR. Maybe we should rename this param to "rviz_default_planner_config" to minimize confusion.

rhaschke committed:
load default_planner_config from default location

instead of loading from `/<ns>/default_planner_config`, use
`/<ns>/move_group/<group>/default_planner_config`, which is the default
location for `planner_configs` too

rhaschke commented:
My adaption loads the `default_planner_config` from the `move_group's` namespace, considering also the selected group when there is a configuration for it. Hence in `ompl_planning.yaml`, one can now specify:

``` yaml
arm:
  default_planner_config: RRTConnectkConfigDefault
  planner_configs:
    - RRTkConfigDefault
    - RRTConnectkConfigDefault
```

Obviously, one can overwrite that setting manually in launch files - after loading the yaml defaults.

The same default could be used by the `move_group` node too. 
However, there is a heuristic to choose a config if I remember correctly.
@k-okada Do you like that modification?

rhaschke committed:
new method MoveGroup::getDefaultPlannerId(const std::string &group)

... to retrieve default planner config from param server
moved corresponding code from rviz plugin to MoveGroup interface
to facilitate re-use

sachinchitta commented:
Nice change! @rhaschke
